src/Task/Occurrence.ts

```diff
@@ -0,0 +1,124 @@
+import type { Moment } from 'moment/moment';
+import { compareByDate } from '../lib/DateTools';
+
+export class Occurrence {
```

Comment:
I think it would be worth just a little JSdoc here, at least saying what an occurrence is and mentioning that this is an implementation detail of Recurrence.

Reply:
I'm going to resolve this and merge it anyway, in the hope it gets the SonarLint duplication score down below 3.0% to fix the failing builds.